src/mgmt/srv_target.c

```diff
@@ -1078,7 +1078,7 @@ ds_mgmt_hdlr_tgt_create(crt_rpc_t *tc_req)
 		D_ASSERT(dss_tgt_nr > 0);
 		uuid_copy(vpa.vpa_uuid, tc_in->tc_pool_uuid);
 		/* A zero size accommodates the existing file */
-		vpa.vpa_scm_size = 0;
```

Comment:
See comments above, 0 is set on purpose here to handle existing file, why we change it to tc_meta_size here?

Reply:
The idea is that when we set meta_size in pool create call it should be used here as meta blob size but if it's unset then the meta_size will be the zero value "0" and meta blob size will be the same as the VOS file size. that incorrect?

Reply:
Initially I thought we would need to pass meta_size directly into vos_pool_create which would result in API change (and a cascade of changes see tanabarr/control-mdonssd-pool-create-blobsizes-2-bak branch for details). Now I don't think we need any vos_pool_create changes, explanation below.

When we call into tgt_vos_create_one and set `srv_target.c L1081 vpa.vpa_scm_size = tc_in->tc_meta_size;` we currently have two situations:
1. `vpa.vpa_scm_size == 0` which will result in metablobsz == vosfilesz, this will occur when metablobsz has not been specified in drpc pool create.
2. `vpa_scm_size > 0` which will happen when metablobsz has been specified in drpc pool create and in this case results in metablobsz != vosfilesz.

In tgt_vos_create_one(), vos_pool_create() is called with the passed vpa_scm_size (either zero or metablobsz) and further down the vos_pool_create call stack if scm_sz is zero then the fallocated file size is used but otherwise the new metablobsz is used

vos_pmemobj_create() sets meta_sz = scm_sz if non-zero (if scm_sz is zero the fallocated vos file size is used for meta_sz) so we end up with the metablobsz being used as meta_sz in call to bio_mc_create() which is what we want.

I don't think we need API changes to vos_pool_create chain but I may well be missing something.

Reply:
Current vos_pool_create() supports two modes:
Mode 1: specifying 0 scm_sz, that means the VOS file has to be pre-allocated, and this mode is used for DAOS regular pool creation.
Mode 2: specifying non-zero scm_sz, that means the VOS file will be automatically created, and this mode is used for RDB, sysdb and some tests.

Looks your proposal is to support only mode 1 in the future? I'm ok with it, but not sure if it could cause more code changes.

Reply:
From my understanding the two modes should be supported with @tanabarr proposal.
Mode 1: The end user do not define the meta size.
Mode 2: The end user define the meta size.
@NiuYawei and @tanabarr , Am I missing something ?

Reply:
So that was my understanding @knard-intel but I was unaware of the other aspect that @NiuYawei is alluding to in his comment above. 
In order to maintain the original usage model with the 2 modes for VOS file (pre-allocated-regular and automatically-created) in addition to supplying a custom meta-blob size looks like we need to add the meta-size parameter to vos_pool_create() and make the various API updates. @NiuYawei does that sound sensible?

Reply:
added parameter to vos_pool_create internal API